From a478485d60fae569a0543c44be1aa7f9e5df03a0 Mon Sep 17 00:00:00 2001
From: Stefano Babic <sbabic@denx.de>
Date: Sun, 5 Aug 2012 12:18:53 +0200
Subject: [PATCH 3/3] MX5: mx53loco: do not overwrite the console

On this board, the console is always set to the serial line.
Do not allow to overwrite it when video is enabled.

Signed-off-by: Stefano Babic <sbabic@denx.de>
CC: Fabio Estevam <fabio.estevam@freescale.com>
---
 board/freescale/mx53loco/mx53loco.c |   12 ++++++------
 include/configs/mx53loco.h          |    2 ++
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/board/freescale/mx53loco/mx53loco.c b/board/freescale/mx53loco/mx53loco.c
index cbdcfad..7cfb3f4 100644
--- a/board/freescale/mx53loco/mx53loco.c
+++ b/board/freescale/mx53loco/mx53loco.c
@@ -495,14 +495,14 @@ int print_cpuinfo(void)
 	return 0;
 }
 
-#ifdef CONFIG_BOARD_LATE_INIT
-int board_late_init(void)
+/*
+ * Do not overwrite the console
+ * Use always serial for U-Boot console
+ */
+int overwrite_console(void)
 {
-	setenv("stdout", "serial");
-
-	return 0;
+	return 1;
 }
-#endif
 
 int board_init(void)
 {
diff --git a/include/configs/mx53loco.h b/include/configs/mx53loco.h
index d446543..2d37bcf 100644
--- a/include/configs/mx53loco.h
+++ b/include/configs/mx53loco.h
@@ -231,6 +231,8 @@
 #define CONFIG_VIDEO_IPUV3
 #define CONFIG_CFB_CONSOLE
 #define CONFIG_VGA_AS_SINGLE_DEVICE
+#define CONFIG_SYS_CONSOLE_IS_IN_ENV
+#define CONFIG_SYS_CONSOLE_OVERWRITE_ROUTINE
 #define CONFIG_VIDEO_BMP_RLE8
 #define CONFIG_SPLASH_SCREEN
 #define CONFIG_BMP_16BPP
-- 
1.7.10.4

